sunpy/map/mapbase.py

```diff
@@ -1213,7 +1213,7 @@ def observer_coordinate(self):
                 # representation)
                 return SkyCoord(sc.replicate(rsun=self._rsun_meters(sc.radius)))
 
-            elif any(meta_list) and not set(keys).isdisjoint(self.meta.keys()):
+            elif not any(meta_list) and set(keys).isdisjoint(self.meta.keys()):
```

Comment:
I think this is correct?

Reply:
Clearly not.

Reply:
I think this line should be dropped and we work out the missing keys all time instead?

Reply:
Tweaked now.